jetbrains-core/tst/software/aws/toolkits/jetbrains/services/cloudwatch/logs/insights/QueryingLogGroupsTest.kt

```diff
@@ -0,0 +1,108 @@
+// Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package software.aws.toolkits.jetbrains.services.cloudwatch.logs.insights
+
+import com.intellij.testFramework.RunsInEdt
+import com.intellij.testFramework.runInEdtAndWait
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+import software.aws.toolkits.jetbrains.utils.rules.JavaCodeInsightTestFixtureRule
+import org.assertj.core.api.Assertions.assertThat
+import software.aws.toolkits.resources.message
+import java.util.Calendar
+import java.util.Date
+
+@RunsInEdt
+class QueryingLogGroupsTest {
+    @JvmField
+    @Rule
+    val projectRule = JavaCodeInsightTestFixtureRule()
+    private val validator = QueryEditorValidator
+    private lateinit var view: QueryEditor
+
+    @Before
+    fun setup() {
+        val project = projectRule.project
+        runInEdtAndWait {
+            view = QueryEditor(project)
+        }
+    }
+
+    @Test
+    fun `Absolute or relative time selected`() {
+        getViewDetails(absoluteTime = false, relativeTime = false)
+        assertThat(validator.validateEditorEntries(view)?.message).contains(message("cloudwatch.logs.validation.timerange"))
+    }
+
+    @Test
+    fun `Start date must be before end date`() {
+        val cal = Calendar.getInstance()
+        cal.add(Calendar.DATE, -1)
+        getViewDetails(absoluteTime = true, startDate = Calendar.getInstance().time, endDate = cal.time)
+        assertThat(validator.validateEditorEntries(view)?.message).contains(message("cloudwatch.logs.compare.start.end.date"))
+    }
+
+    @Test
+    fun `Relative Time, no time entered`() {
+        getViewDetails(relativeTime = true, relativeTimeNumber = "")
+        assertThat(validator.validateEditorEntries(view)?.message).contains(message("cloudwatch.logs.no_relative_time_number"))
+    }
+
+    @Test
+    fun `Neither Search Term nor Querying through log groups selected`() {
+        getViewDetails(relativeTime = true)
+        assertThat(validator.validateEditorEntries(view)?.message).contains(message("cloudwatch.logs.no_query_selected"))
+    }
+
+    @Test
+    fun `No search term entered`() {
+        getViewDetails(relativeTime = true, querySearch = true, searchTerm = "")
+        assertThat(validator.validateEditorEntries(view)?.message).contains(message("cloudwatch.logs.no_term_entered"))
+    }
+
+    @Test
+    fun `No query entered`() {
+        getViewDetails(relativeTime = true, queryLogs = true, query = "")
+        assertThat(validator.validateEditorEntries(view)?.message).contains(message("cloudwatch.logs.no_query_entered"))
+    }
+
+    @Test
+    fun `All form entries validated Variant1`() {
```

Comment:
The test names need to describe what they are testing. The difference between this test and the one below it should be obvious from the name